lib/node_modules/@stdlib/math/base/special/factorial2/test/test.js

```diff
@@ -0,0 +1,132 @@
+var incrspace = require( '@stdlib/math/generics/utils/incrspace' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var PINF = require( '@stdlib/math/constants/float64-pinf' );
+var NINF = require( '@stdlib/math/constants/float64-ninf' );
+var EPS = require( '@stdlib/math/constants/float64-eps' );
+var factorial2 = require( './../lib' );
+
+
+// FIXTURES //
+
+var integers = require( './fixtures/julia/integers.json' );
+var decimals = require( './fixtures/julia/decimals.json' );
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof factorial2, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'if provided a negative integer, the function returns `NaN`', function test( t ) {
+	var values = incrspace( -1.0, -1000.0, -1.0 );
+	var v;
+	var i;
+
+	for ( i = 0; i < values.length; i++ ) {
+		v = factorial2( values[ i ] );
+		t.equal( isnan( v ), true, 'returns NaN when provided ' + values[ i ] );
+	}
+	t.end();
+});
+
+tape( 'if provided negative infinity, the function returns `NaN`', function test( t ) {
+	var v = factorial2( NINF );
+	t.equal( isnan( v ), true, 'returns NaN when provided negative infinity' );
+	t.end();
+});
+
+tape( 'if provided positive infinity, the function returns `+infinity`', function test( t ) {
+	var v = factorial2( PINF );
+	t.ok( v, PINF, 'returns +infinity when provided +infinity' );
+	t.end();
+});
+
+tape( 'if provided `NaN`, the function returns `NaN`', function test( t ) {
+	var v = factorial2( NaN );
+	t.equal( isnan( v ), true, 'returns NaN when provided a NaN' );
+	t.end();
+});
+
+tape( 'if `x > 170.6144...`, the function returns positive infinity', function test( t ) {
+	var values = incrspace( 170.615, 1000.0, 10.1234 );
+	var v;
+	var i;
+
+	for ( i = 0; i < values.length; i++ ) {
+		v = factorial2( values[ i ] );
+		t.equal( v, PINF, 'returns +infinity when provided ' + values[ i ] );
+	}
+	t.end();
+});
+
+tape( 'if `x < -171.56749...`, the function returns positive infinity', function test( t ) {
+	var values = incrspace( -171.57, -1000.0, -10.1234 );
+	var v;
+	var i;
+
+	for ( i = 0; i < values.length; i++ ) {
+		v = factorial2( values[ i ] );
+		t.equal( v, PINF, 'returns +infinity when provided ' + values[ i ] );
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the double factorial function (positive integers < 171)', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = integers.x;
+	expected = integers.expected;
+	for ( i = 0; i < x.length; i++ ) {
+		v = factorial2( x[ i ] );
+		if ( v === expected[ i ] ) {
+			t.equal( v, expected[ i ], 'returns '+expected[i]+' when provided '+x[i] );
+		} else {
+			delta = abs( v - expected[ i ] );
+			tol = EPS * Math.max( 1.0, abs( v ), abs( expected[ i ] ) );
+			t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Tolerance: ' + tol + '.' );
```

Comment:
Can you replace the last two lines with 

``` javascript
tol = EPS * abs( expected[ i ] );
t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. Value: '+v+'. E: '+expected[i]+'. Δ: '+delta+'. tol: '+tol );
```

and check whether the tests still pass?